### PR TITLE
docs: Cluster overview page

### PR DESCRIPTION
**Pull request by kostis-codefresh**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Ongoing work for https://github.com/argoproj/argo-cd/issues/5635

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [X] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: docs/basics/clusters/index.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Argo CD target clusters
+
+By default an Argo CD instance can deploy applications to the [same cluster](internal/) it is installed on. However, Argo CD has the capability to connect and deploy to [external clusters](external).
```
Comment (todaywasawesome) [inline docs/basics/clusters/index.md:3]:
```suggestion
By default an Argo CD instance can deploy applications to the [same cluster](internal/) it is installed on. Additionally, Argo CD has the capability to connect and deploy to [external clusters](external).
```

The use of "however" seems in opposition and I think it's better replaced with "Additionally"

File: docs/basics/clusters/index.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+Disadvantages
+
+* Single point of failure
+* Target cluster API’s must be accessible to the central instance
+* Single attack surface
+* RBAC and SSO might not match organizational requirements
+* Impossible to upgrade different clusters in waves
+* No separation between QA/Staging/Prod installations
+* Argo CD performance may degrade with many applications and clusters as they scale
```
Comment (todaywasawesome) [inline docs/basics/clusters/index.md:62]:
```suggestion
* Argo CD performance may degrade with many applications and clusters as they scale - see [High Availability](../../operator-manual/high_availability/) for more information.
```

File: docs/basics/clusters/index.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+![Hybrid topology 1](../../assets/hybrid1.png)
+
+In this example there are 3 management Argo CD instances. This allows the Production instance to have different constraints than the non-production instances while also making upgrades of Argo CD itself more granular (Production instance can be updated at the end after QA and staging).
+
+Example 2 - Mix of Argo CD instances per region and environment
+
+![Hybrid topology 2](../../assets/hybrid2.png)
+
+In this example there are different Argo CD instances per region that manage other clusters, while each team gets also its own Argo CD instance. This allows developer teams to have their own config for their ArgoCD instance and also there is no fear of affecting other teams during upgrades.
```
Comment (todaywasawesome) [inline docs/basics/clusters/index.md:86]:
```suggestion
In this example there are different Argo CD instances per region that manage other clusters, while each team gets also its own Argo CD instance. This allows developer teams to have their own config for their Argo CD instance and also there is no fear of affecting other teams during upgrades.
```

File: docs/basics/clusters/index.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+## Which strategy to choose
+
+There is no right or wrong Argo CD topology. Each organization has different needs. It is also possible to start with one topology and change to another as the organization needs are changing.
+
+As a starting point we suggest we have at least 2 ArgoCD categories (which might be individual instances or multiple installations)
```
Comment (todaywasawesome) [inline docs/basics/clusters/index.md:92]:
```suggestion
As a starting point we suggest we have at least 2 Argo CD categories (which might be individual instances or multiple installations)
```

File: docs/basics/clusters/index.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
+# Argo CD target clusters
+
+By default an Argo CD instance can deploy applications to the [same cluster](internal/) it is installed on. However, Argo CD has the capability to connect and deploy to [external clusters](external).
+
+This means that there are several topologies for handling multiple clusters with Argo CD
+
+1. Using one Argo CD instance per cluster
+1. Using a centralized Argo CD instance that handles all cluster
```
Comment (blakepettersson) [inline docs/basics/clusters/index.md:8]:
```suggestion
1. Using a centralized Argo CD instance that handles all clusters
```

File: docs/basics/clusters/index.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+Disadvantages
+
+* Single point of failure
+* Target cluster API’s must be accessible to the central instance
+* Single attack surface
+* RBAC and SSO might not match organizational requirements
+* Impossible to upgrade different clusters in waves
+* No separation between QA/Staging/Prod installations
+* Argo CD performance may degrade with many applications and clusters as they scale
+* In theory requires less resources per each deployment cluster (as it only runs apps and not ArgoCD itself)
```
Comment (blakepettersson) [inline docs/basics/clusters/index.md:63]:
```suggestion
* In theory requires less resources per deployment cluster (as it only runs apps and not Argo CD itself)
```

File: docs/basics/clusters/index.md (hunk 3 above)
Comment (blakepettersson) [inline docs/basics/clusters/index.md:86]:
```suggestion
In this example there are different Argo CD instances per region that manage other clusters, while each team also gets its own Argo CD instance. This allows developer teams to have their own config for their Argo CD instance and there is also no fear of affecting other teams during upgrades.
```

File: docs/basics/clusters/index.md (hunk 4 above)
Comment (blakepettersson) [inline docs/basics/clusters/index.md:92]:
```suggestion
As a starting point we suggest to have at least 2 Argo CD categories (which might be individual instances or multiple installations)
```

**Review by todaywasawesome — APPROVED** [review]

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
@kostis-codefresh can you run the images through squoosh.app or similar? The files are a bit big right now.

File: docs/basics/clusters/index.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
+# Argo CD target clusters
```
Comment (crenshaw-dev) [inline docs/basics/clusters/index.md:1]: Should this file be added to mkdocs.yml so it shows up in the ToC?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Comment by kostis-codefresh** [comment]
@crenshaw-dev I converted the images to jpg using squoosh.app per your suggestion and they are much smaller now.

The page is already added in mkdocs. This doc page is for the new documentation (notice the target is of the pull request is the refresh-docs branch and not master)

Entry https://github.com/argoproj/argo-cd/blob/refresh-docs/mkdocs.yml#L41
How it looks https://argo-cd--11637.org.readthedocs.build/en/11637/basics/clusters/

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**